core/foundation/test/testStringUtils.cxx

```diff
@@ -20,4 +20,23 @@ TEST(StringUtils, Split)
    test(",,,", {}, true);
    test(",,a", {"", "", "a"}, false);
    test("", {""}, false);
+
+}
+
+TEST(StringUtils, Join)
+{
+   // Test that ROOT::Join behaves like str.join from Python.
+
+   auto test_join = [](const std::string& ref, const std::string& sep, const std::vector<std::string> &strings) {
+      auto out = ROOT::Join(sep, strings);
+      EXPECT_EQ(out, ref) << "ROOT::Join gave wrong result.";
+   };
+   test_join("apple,orange,banana", ",", {"apple", "orange", "banana"});
```

Comment:
The usage of these tests is different from the function signature. Signature is
```cpp
inline std::string Join(const std::string& sep, T&&...strings) {
```

which means the usage should be `Join(",", "a","b","c")` as I understand it, not `Join(",", {"a", "b", "c"})`

Reply:
btw, we can totally also add `Join(std::string, std::vector<std::string>)` signature, that can go in a separate PR (not necessarily done by you ofc :) )

Reply:
Yep, but then I was confused on how to define the test lambda. I think it needs c++20: https://stackoverflow.com/questions/3575901/can-lambda-functions-be-templated

Reply:
Concerning std::vector, I can add it, sure. Maybe via an unpack_vector helper function? https://stackoverflow.com/questions/41980888/how-to-convert-from-stdvector-to-args

Reply:
One doesn't need to be implemented in terms of the other https://godbolt.org/z/e7fKojY59